### DISCUSSION TITLE
best way to do things like 'git gc'

### DISCUSSION THREAD

**Discussion by bdklahn** [gitpython-developers/GitPython#1174] [Q&A] [answered]
I've already read "Using git directly".
But using IPython object introspection, it doesn't look like something like "gc" is a method there.

The following seems to work. Is that the recommended way?

a_repo.git.execute("git gc".split())

**Comment by Byron** [comment, marked as answer]
IPython is unable to see what's possible as these options are generated on the fly.

Thus invoking `a_repo.git.gc()` would be the idiomatic way to achieve that.

Please feel free to keep commenting despite the issue being closed in case there is a follow up.

**Comment by bdklahn** [comment]
Ah.
Thanks!
And that, presumably, is also why IPython TAB completion didn't complete to ".  . . .gc", when I tried that.

So what might the "split=true" be in yours? I did a string split in mine, because the `execute` command seemed to want an iterable (like from space-separated command line arguments, I presume), rather than a string. But with `a_repo.git.gc . . .` there would be no additional arguments.

**Comment by Byron** [comment]
Great to hear. The argument was a mistake.
